### PR TITLE
search: rename most jobs to be consistent

### PR DESCRIPTION
**Pull request by keegancsmith**
This is series of refactors which just rename the names of jobs. Additionally in some cases we updated the return value of the Name() function. The philosophy guiding this change is:

- Avoid stutter (`zoekt.Zoekt*` -> `zoekt.*`)
- Mention Text if it is a text search job
- Mention Search if it is a search job (vs alert jobs/etc)
- Name should be `Upper(package) + TypeName`

Complete version of https://github.com/sourcegraph/sourcegraph/pull/35935

Test Plan: all just refactors, so just that tests pass.

### CODE REVIEW

File: internal/search/job/jobutil/mapper.go [hunk 1]
```diff
@@ -18,16 +18,16 @@ type Mapper struct {
...
+	MapZoektSymbolSearchJob       func(*zoekt.SymbolSearchJob) *zoekt.SymbolSearchJob
+	MapZoektGlobalSymbolSearchJob func(*zoekt.GlobalSymbolSearchJob) *zoekt.GlobalSymbolSearchJob
+	MapSearcherJob                func(*searcher.TextSearchJob) *searcher.TextSearchJob
+	MapSymbolSearcherJob          func(*searcher.SymbolSearchJob) *searcher.SymbolSearchJob
 	MapRepoSearchJob              func(*run.RepoSearchJob) *run.RepoSearchJob
-	MapRepoUniverseTextSearchJob  func(*zoekt.ZoektGlobalSearchJob) *zoekt.ZoektGlobalSearchJob
-	MapStructuralSearchJob        func(*structural.StructuralSearchJob) *structural.StructuralSearchJob
-	MapCommitSearchJob            func(*commit.CommitSearchJob) *commit.CommitSearchJob
-	MapComputeExcludedReposJob    func(*repos.ComputeExcludedReposJob) *repos.ComputeExcludedReposJob
+	MapRepoUniverseTextSearchJob  func(*zoekt.GlobalTextSearchJob) *zoekt.GlobalTextSearchJob
```
Comment (tbliu98) [inline internal/search/job/jobutil/mapper.go:27]:
```suggestion
	MapZoektGlobalTextSearchJob  func(*zoekt.GlobalTextSearchJob) *zoekt.GlobalTextSearchJob
```

File: internal/search/job/jobutil/mapper.go [hunk 2]
```diff
@@ -18,16 +18,16 @@ type Mapper struct {
 	MapJob func(job job.Job) job.Job
 
 	// Search Jobs (leaf nodes)
-	MapZoektRepoSubsetSearchJob   func(*zoekt.ZoektRepoSubsetSearchJob) *zoekt.ZoektRepoSubsetSearchJob
-	MapZoektSymbolSearchJob       func(*zoekt.ZoektSymbolSearchJob) *zoekt.ZoektSymbolSearchJob
-	MapZoektGlobalSymbolSearchJob func(*zoekt.ZoektGlobalSymbolSearchJob) *zoekt.ZoektGlobalSymbolSearchJob
-	MapSearcherJob                func(*searcher.SearcherJob) *searcher.SearcherJob
-	MapSymbolSearcherJob          func(*searcher.SymbolSearcherJob) *searcher.SymbolSearcherJob
+	MapZoektRepoSubsetSearchJob   func(*zoekt.RepoSubsetTextSearchJob) *zoekt.RepoSubsetTextSearchJob
```
Comment (tbliu98) [inline internal/search/job/jobutil/mapper.go:21]:
```suggestion
	MapZoektRepoSubsetTextSearchJob   func(*zoekt.RepoSubsetTextSearchJob) *zoekt.RepoSubsetTextSearchJob
```

**Review by tbliu98 — APPROVED** [review]
Just a couple nits on keeping `Mapper` field names consistent as well. 
I feel like I want those to continue to match their associated jobs' names as closely as possible but I'm not sure what's best for the Searcher jobs. 

`MapSearcherJob` --> `MapSearcherTextSearchJob` ?
`MapSymbolSearcherJob` --> `MapSearcherSymbolSearchJob` ?

I don't really have a strong opinion either way on those, just throwing it out there

**Review by rvantonder — APPROVED** [review]

File: enterprise/internal/codemonitors/search.go [hunk 3]
```diff
@@ -220,15 +220,15 @@ func addCodeMonitorHook(in job.Job, hook commit.CodeMonitorHook) (_ job.Job, err
...
+		case *commit.SearchJob:
 			commitSearchJobCount++
 			if commitSearchJobCount > 1 && err == nil {
 				err = ErrInvalidMonitorQuery
 			}
 			jobCopy := *typedAtom
 			jobCopy.CodeMonitorSearchWrapper = hook
 			return &jobCopy
-		case *repos.ComputeExcludedReposJob, *jobutil.NoopJob:
+		case *repos.ComputeExcludedJob, *jobutil.NoopJob:
```
Comment (rvantonder) [inline enterprise/internal/codemonitors/search.go:231]: actually coming back to this change, that I missed in the draft PR: why rename this one? I have always been confused by this job name (Compute excluded _what_?)

Comment (rvantonder) [reply enterprise/internal/codemonitors/search.go:231]: oh never mind, it's prefixed by `repos` now

File: enterprise/internal/codemonitors/search.go [hunk 4]
```diff
@@ -220,15 +220,15 @@ func addCodeMonitorHook(in job.Job, hook commit.CodeMonitorHook) (_ job.Job, err
...
 			// ComputeExcludedReposJob is fine for code monitor jobs
```
Comment (rvantonder) [inline enterprise/internal/codemonitors/search.go:232]:
```suggestion
			// ComputeExcludedJob is fine for code monitor jobs
```

**Comment by keegancsmith** [comment]
updated the mapper fields. I made it so the map job names are `"Map" + type.Name()`.

**Review by camdencheek — COMMENTED** [review]
thanks for the push @keegancsmith!

**Pull request merged**